tools/const_finder/lib/const_finder.dart

```diff
@@ -72,12 +74,42 @@ class _ConstVisitor extends RecursiveVisitor<void> {
     super.visitConstructorInvocation(node);
   }
 
+  @override
+  void visitClass(Class node) {
+    // check if this is a class that we should ignore
+    if (_isStaticIconProvider(node)) {
+      inIgnoredClass = true;
+      super.visitClass(node);
+      inIgnoredClass = false;
+      return;
+    }
+    // not an ignored class
+    super.visitClass(node);
```

Comment:
Flip the sense and unindent the larger portion.

```suggestion
    if (!_isStaticIconProvider(node)) {
      // Not an ignored class.
      super.visitClass(node);
      return;
    }
    inIgnoredClass = true;
    super.visitClass(node);
    inIgnoredClass = false;
```